File: backend/source/core/entities/Color.ts
import { Cielab } from "../valueObjects/Cielab";
import { Hexadecimal } from "../valueObjects/Hexadecimal";
import { RGB } from "../valueObjects/RGB";

export class Color {
    constructor(
        readonly HexadecimalCode: Hexadecimal,
        readonly RGBCode: RGB,
        readonly CielabCode: Cielab,
    ) { }

    public Distance(color: Color): number {
        return this.CielabCode.CalculateCie94Distance(color.CielabCode);
    }

    public Equals(color: Color) {
        return this.HexadecimalCode.ToString() == color.HexadecimalCode.ToString();
    }
}

File: backend/aspnetcore_backend/Core.Domain/Entities/Color.cs
using Core.Domain.ValueObjects;

namespace Core.Domain.Entities
{
    public class Color
    {
        public Hexadecimal HexadecimalCode { get; }
        public RGB RGBCode { get; }
        public Cielab CielabCode { get; }

        public float Distance(Color color) => this.CielabCode.CalculateCie94Distance(color.CielabCode);

    }
}
